### PR TITLE
Refresh GC reliability scripts/tests.

### PR DESCRIPTION
**Pull request by VSadov**
* GC reliability tests can build on Linux
* GC reliability tests can run on Linux
* GC reliability tests are buildable together with other Pri-0 tests. 
( BuildOnly though, we do not want to run them with other tests )

Verified running locally on  Windows, Linux, x64, arm64 

next step (possibly in a separate PR) - adding trigger commands to run in the lab.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -256,6 +263,10 @@ jobs:
         - Windows_NT_x64
         - Windows_NT_x86
         helixQueueGroup: all
+      ${{ if eq(variables['Build.DefinitionName'], 'coreclr-gc-reliability-chk') }}:
+        platforms:
+        - Windows_NT_x64
+        helixQueueGroup: all
```
Comment (echesakovMSFT) [inline azure-pipelines.yml]: I would avoid using `all` for `helixQueueGroup` this could block all the queues for hours from running - just use `ci`.

**Comment by echesakovMSFT** [comment]
@VSadov You will need to create a build defition through UI. I assume your intention is to run these test on Release product build? Do you want to make them a part of public CI or internal? If last one, we should discuss this

**Comment by VSadov** [comment]
We will require external contributors to run reliability tests, so I guess it should be runnable on public PRs. 
It does not need to be "triggerable" externally, if that is even possible/makes sense.

**Pull request closed without merging**